### ISSUE TITLE
`ollama push` and `ollama pull` are slow or hang on windows

### ISSUE THREAD

**Issue by ewebgh33** [ollama/ollama#2850]
Can't download ANY models.

What is happening? Not my internet, speed test blasts. 
Your servers OK?

Happening on Windows version buggy still? Using latest, 0.1.27 (Win11).

As per docs, set Windows environment variable to:
OLLAMA_MODELS = D:\AI\text\ollama-models
I am familiar with environment variables and this worked with llama2 a few days ago.

Now in Powershell
`ollama pull phind-codellama`
Says will take 99hrs, has downloaded 82kb
Then quits DL
`Error: context canceled`

Just tried codellama:70b, same thing. 99hrs, cancels with error.

This is why people ask, "why can't we just use a GGUF" or AWQ or whatever. There are multiple sources that host the models, but we need these hashed files and blobs that only Ollama has. Centralised models, point of failure, case in point this ticket.
Some models I already have as I run them in Oobabooga. Is it too much to ask I could use these already downloaded (and large) models? And not need two copies of the same thing, albeit in different formats.

Rebooted - no change. Can't download ANY models.

**Comment by EmmaWebGH** [comment]
This is still happening today, I tried again and it got to 75kb (total downloaded, not speed) before it got so slow that it errored out and killed itself.

**Comment by AstralTomate** [comment]
Turn off the windows firewall or make a rule exception. That worked for me.

**Comment by EmmaWebGH** [comment]
I managed to download codellama34b a short time back but the downloading is hella buggy, seems to fail most of the time and then luckily just succeed randomly.

I would not have thought that firewall was an issue, as previously I was able to download anything and everything via Ollama running in WSL. It's the Windows native version that is giving me issues. I've never had problems downloading anything really - just this, now.

**Comment by dhiltgen** [comment]
I'm sorry you're having trouble downloading models.  We had a few glitches last week on the hub although I don't think those would explain what you're seeing.  Are you still experiencing slow downloads?  Did changing firewall settings have any impact?  Do you have any 3rd party AV software on your system that might be doing packet inspection?

If you're still having problems, could you run the server with OLLAMA_DEBUG="1" set and share the logs when you're trying to download and seeing the extremely slow throughput?

We're working on some improvements to throttling the download to try to optimize for the available bandwidth in https://github.com/ollama/ollama/pull/2221 which may help.

**Comment by SarkarKurdish** [comment]
> Turn off the windows firewall or make a rule exception. That worked for me.

Worked for me too, thanks

**Comment by hermitbernard** [comment]
Turned off the firewall. Worked to pull 1 model, then stopped working if I try to download other models.

**Comment by MarsThunder** [comment]
First time I experienced this. Happened today. Pulling llava and solar.  First it has speed of 7B/sec (molasses).  So I Ctrl+C and then try again. Usually, it will then download at normal speed.  If I don't get the ASCII white progress bar within a second, it will fail.  Once I pull and see that progress bar within a second, it seems to be normal.  I am chalking it up to busy download server and AI madness where everyone is downloading!

**Comment by bmizerany** [comment]
Hi everyone,

We're actively working on slow downloads for all. Please help us test a new experimental solution. Instructions are here: https://github.com/ollama/ollama/issues/1736#issuecomment-2102983113

Please, please, please give us feedback in that issue if you can!

**Comment by supasympa** [comment]
I am experiencing this. It seems to trash my local network somehow

**Comment by abshkd** [comment]
I am not able to push either. it goes 98% then slows to modem speed, its hsowing as done but is just stuck there at 
`pushing...100%....` As a backup i am just dumping to huggingface.

**Comment by thedarkfalcon** [comment]
I am experiencing random slow down to a crawls via the application. It was downloading at around 10MB/s, then randomly slowed down to a crawl of a few kilobytes a second.

<img width="880" height="94" alt="Image" src="https://github.com/user-attachments/assets/01b77bd5-2578-455e-8772-9a7e4425bd13" />

If I start downloading the file directly through my web browser I again get the expected speed:

<img width="1280" height="431" alt="Image" src="https://github.com/user-attachments/assets/5bf935ab-da12-499f-87a9-120a2fb3f2f6" />

FYI I got the direct download link from [this website](https://ollama-direct-downloader.vercel.app/).

Stopping and starting the download seemed to fix the problem:

<img width="890" height="124" alt="Image" src="https://github.com/user-attachments/assets/5ff3056f-e069-41dd-b4f0-9b21c6c3d58c" />

**Comment by MarsThunder** [comment]
I just checked this morning to test.  I did a pull instead of run and I got consistent 27-33MB/s.

<img width="1313" height="268" alt="Image" src="https://github.com/user-attachments/assets/bf193f03-905b-4225-8525-e824ff25bc78" />

Wonder if it is the version installed?  I am using version 0.9.6, but on Ubuntu 24.04.  I just deleted and tried 'run' and it was averaging 35MB/s.  Maybe it was time of day when server was busy?
